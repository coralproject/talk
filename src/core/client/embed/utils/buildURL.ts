export default function buildURL({
  protocol = window.location.protocol,
  hostname = window.location.hostname,
  port = window.location.port,
  pathname = window.location.pathname,
  search = window.location.search,
  hash = window.location.hash,
} = {}) {
  if (search && search[0] !== "?") {
    search = `?${search}`;
  } else if (search === "?") {
    search = "";
  }
  return `${protocol}//${hostname}${
    port ? `:${port}` : ""
  }${pathname}${search}${hash}`;
}
