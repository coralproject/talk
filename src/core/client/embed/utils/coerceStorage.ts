import { createInMemoryStorage } from "coral-framework/lib/storage";

function coerceStorage(type: "localStorage" | "sessionStorage"): Storage {
  let storage: Storage;
  try {
    storage = window[type];
  } catch {
    // eslint-disable-next-line no-console
    console.debug(`${type} not available, use in-memory replacement`);

    storage = createInMemoryStorage();
  }

  return storage;
}

export default coerceStorage;
