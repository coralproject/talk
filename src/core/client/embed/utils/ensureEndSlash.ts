export default function ensureEndSlash(p: string) {
  return p.match(/\/$/) ? p : `${p}/`;
}
