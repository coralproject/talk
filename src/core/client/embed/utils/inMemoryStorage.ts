class InMemoryStorage implements Storage {
  private readonly data = new Map<string, string>();

  public get length() {
    return this.data.size;
  }

  public clear() {
    this.data.clear();
  }

  public key(n: number) {
    const keys = [...this.data.keys()];
    if (keys.length < n || n < 0) {
      return null;
    }

    return keys[n];
  }

  public getItem(key: string): string | null {
    return this.data.get(key) || null;
  }

  public removeItem(key: string): void {
    this.data.delete(key);
  }

  public setItem(key: string, value: string): void {
    this.data.set(key, value);
  }
}

export default InMemoryStorage;
