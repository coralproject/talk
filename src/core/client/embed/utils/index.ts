export { default as buildURL } from "./buildURL";
export { default as ensureEndSlash } from "./ensureEndSlash";
export { default as ensureNoEndSlash } from "./ensureNoEndSlash";
export { default as startsWith } from "./startsWith";
export { default as prefixStorage } from "./prefixStorage";
export { default as parseHashQuery } from "./parseHashQuery";
