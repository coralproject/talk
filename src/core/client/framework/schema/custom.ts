/**
 * TODO: (cvle) This file is a workaround to have Relay compatible enum types.
 * This should be generated by `graphql-schema-typescript`.
 */

import { RelayEnumLiteral } from "../types";
import {
  GQLCOMMENT_FLAG_DETECTED_REASON,
  GQLCOMMENT_FLAG_REASON,
  GQLCOMMENT_FLAG_REPORTED_REASON,
  GQLCOMMENT_SORT,
  GQLCOMMENT_STATUS,
  GQLLOCALES,
  GQLMODERATION_MODE,
  GQLSTORY_STATUS,
  GQLUSER_AUTH_CONDITIONS,
  GQLUSER_ROLE,
  GQLUSER_STATUS,
} from "./__generated__/types";

export type GQLUSER_ROLE_RL = RelayEnumLiteral<typeof GQLUSER_ROLE>;
export type GQLUSER_STATUS_RL = RelayEnumLiteral<typeof GQLUSER_STATUS>;
export type GQLCOMMENT_FLAG_DETECTED_REASON_RL = RelayEnumLiteral<
  typeof GQLCOMMENT_FLAG_DETECTED_REASON
>;
export type GQLCOMMENT_FLAG_REASON_RL = RelayEnumLiteral<
  typeof GQLCOMMENT_FLAG_REASON
>;
export type GQLCOMMENT_FLAG_REPORTED_REASON_RL = RelayEnumLiteral<
  typeof GQLCOMMENT_FLAG_REPORTED_REASON
>;
export type GQLCOMMENT_SORT_RL = RelayEnumLiteral<typeof GQLCOMMENT_SORT>;
export type GQLCOMMENT_STATUS_RL = RelayEnumLiteral<typeof GQLCOMMENT_STATUS>;
export type GQLLOCALES_RL = RelayEnumLiteral<typeof GQLLOCALES>;
export type GQLMODERATION_MODE_RL = RelayEnumLiteral<typeof GQLMODERATION_MODE>;
export type GQLSTORY_STATUS_RL = RelayEnumLiteral<typeof GQLSTORY_STATUS>;
export type GQLUSER_AUTH_CONDITIONS_RL = RelayEnumLiteral<
  typeof GQLUSER_AUTH_CONDITIONS
>;
