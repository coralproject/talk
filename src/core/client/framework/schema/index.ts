export * from "./__generated__/types";
export * from "./custom";
