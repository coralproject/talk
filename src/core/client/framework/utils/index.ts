export { default as buildURL } from "./buildURL";
export { default as parseURL } from "./parseURL";
export { default as modifyQuery } from "./modifyQuery";
