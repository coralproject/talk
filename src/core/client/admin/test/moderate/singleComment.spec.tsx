import { pureMerge } from "coral-common/utils";
import {
  GQLCOMMENT_STATUS,
  MutationToApproveCommentResolver,
  MutationToRejectCommentResolver,
  QueryToCommentResolver,
} from "coral-framework/schema";
import { GQLResolver } from "coral-framework/schema";
import {
  createMutationResolverStub,
  createQueryResolverStub,
  createResolversStub,
  CreateTestRendererParams,
  replaceHistoryLocation,
  toJSON,
  waitForElement,
  within,
} from "coral-framework/testHelpers";

import create from "../create";
import {
  emptyModerationQueues,
  rejectedComments,
  reportedComments,
  settings,
  users,
} from "../fixtures";

const viewer = users.admins[0];

beforeEach(async () => {
  replaceHistoryLocation("http://localhost/admin/moderate");
});

async function createTestRenderer(
  params: CreateTestRendererParams<GQLResolver> = {}
) {
  const { testRenderer, context, subscriptionHandler } = create({
    ...params,
    resolvers: pureMerge(
      createResolversStub<GQLResolver>({
        Query: {
          settings: () => settings,
          viewer: () => viewer,
        },
      }),
      params.resolvers
    ),
    initLocalState: (localRecord, source, environment) => {
      localRecord.setValue(true, "loggedIn");
      if (params.initLocalState) {
        params.initLocalState(localRecord, source, environment);
      }
    },
  });
  return { testRenderer, context, subscriptionHandler };
}

const comment = rejectedComments[0];
const commentStub = createQueryResolverStub<QueryToCommentResolver>(
  ({ variables }) => {
    expectAndFail(variables).toEqual({ id: comment.id });
    return reportedComments[0];
  }
);

beforeEach(() => {
  replaceHistoryLocation(
    `http://localhost/admin/moderate/comment/${comment.id}`
  );
});

it("renders single comment view", async () => {
  const { testRenderer } = await createTestRenderer({
    resolvers: {
      Query: {
        comment: commentStub,
      },
    },
  });
  const { getByTestID } = within(testRenderer.root);
  const container = await waitForElement(() =>
    getByTestID("single-moderate-container")
  );
  expect(toJSON(container)).toMatchSnapshot();
});

it("approves single comment", async () => {
  const approveCommentStub = createMutationResolverStub<
    MutationToApproveCommentResolver
  >(({ variables }) => {
    expectAndFail(variables).toMatchObject({
      commentID: comment.id,
      commentRevisionID: comment.revision.id,
    });
    return {
      comment: {
        id: comment.id,
        status: GQLCOMMENT_STATUS.APPROVED,
        statusHistory: {
          edges: [
            {
              node: {
                id: "mod-action",
                author: {
                  id: viewer.id,
                  username: viewer.username,
                },
              },
            },
          ],
        },
      },
      moderationQueues: emptyModerationQueues,
    };
  });

  const { testRenderer } = await createTestRenderer({
    resolvers: {
      Query: {
        comment: commentStub,
      },
      Mutation: {
        approveComment: approveCommentStub,
      },
    },
  });

  const { getByLabelText, getByTestID } = within(testRenderer.root);
  const ApproveButton = await waitForElement(() => getByLabelText("Approve"));
  ApproveButton.props.onClick();

  expect(
    toJSON(getByTestID(`moderate-comment-${comment.id}`))
  ).toMatchSnapshot();

  expect(approveCommentStub.called).toBe(true);
});

it("rejects single comment", async () => {
  const rejectCommentStub = createMutationResolverStub<
    MutationToRejectCommentResolver
  >(({ variables }) => {
    expectAndFail(variables).toMatchObject({
      commentID: comment.id,
      commentRevisionID: comment.revision.id,
    });
    return {
      comment: {
        id: comment.id,
        status: GQLCOMMENT_STATUS.REJECTED,
        author: comment.author,
        statusHistory: {
          edges: [
            {
              node: {
                id: "mod-action",
                author: {
                  id: viewer.id,
                  username: viewer.username,
                },
              },
            },
          ],
        },
      },
      moderationQueues: emptyModerationQueues,
    };
  });

  const { testRenderer } = await createTestRenderer({
    resolvers: {
      Query: {
        comment: commentStub,
      },
      Mutation: {
        rejectComment: rejectCommentStub,
      },
    },
  });

  const { getByLabelText, getByTestID } = within(testRenderer.root);
  const RejectButton = await waitForElement(() => getByLabelText("Reject"));
  RejectButton.props.onClick();

  expect(
    toJSON(getByTestID(`moderate-comment-${comment.id}`))
  ).toMatchSnapshot();
  expect(rejectCommentStub.called).toBe(true);
});
