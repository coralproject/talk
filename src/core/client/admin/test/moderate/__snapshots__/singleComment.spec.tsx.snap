// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`approves single comment 1`] = `
<div
  className="Card-root ModerateCard-root"
  data-testid="moderate-comment-comment-0"
>
  <div
    className="Box-root Flex-root Flex-flex"
  >
    <div
      className="ModerateCard-mainContainer"
    >
      <div
        className="ModerateCard-topBar"
      >
        <div
          className="Box-root Flex-root Flex-flex Flex-alignCenter"
        >
          <button
            className="BaseButton-root ModerateCard-username"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            <span
              className="Box-root Typography-root Typography-heading4 Typography-colorTextPrimary Username-root"
            >
              Isabelle
            </span>
          </button>
          <time
            className="RelativeTime-root Timestamp-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <button
            className="BaseButton-root FeatureButton-root"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            <span>
              Feature
            </span>
          </button>
        </div>
      </div>
      <div
        className="Box-root Typography-root Typography-bodyCopy Typography-colorTextPrimary ModerateCard-content CommentContent-root"
        dangerouslySetInnerHTML={
          Object {
            "__html": "This is the last random sentence I will be writing and I am going to stop mid-sent",
          }
        }
      />
      <div
        className="ModerateCard-footer"
      >
        <div
          className="Box-root HorizontalGutter-root HorizontalGutter-full"
        >
          <div>
            <a
              className="TextLink-root ModerateCard-link"
              href="http://localhost/comment/0"
              target="_blank"
            >
              View Context
              <span
                aria-hidden="true"
                className="Icon-root Icon-xs TextLink-icon"
              >
                open_in_new
              </span>
            </a>
          </div>
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="Box-root Flex-root Flex-flex Flex-itemGutter gutter"
            >
              <span
                className="Marker-root Marker-colorError Marker-variantRegular"
              >
                <span>
                  Spam
                </span>
                 
                <span
                  className="Count-root"
                >
                  2
                </span>
              </span>
              <button
                aria-controls="uuid-0"
                aria-expanded={false}
                className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Markers-detailsButtonColorRegular Button-variantRegular Markers-detailsButton"
                data-color="regular"
                data-variant="regular"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                type="button"
              >
                <span
                  className="Markers-detailsText"
                >
                  Details
                </span>
                <span
                  aria-hidden="true"
                  className="Icon-root Icon-sm"
                >
                  arrow_drop_down
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ModerateCard-separator ModerateCard-ruledSeparator"
    />
    <div
      className="Box-root Flex-root ModerateCard-aside ModerateCard-asideWithoutReplyTo Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionColumn gutter"
    >
      <div
        className="ModerateCard-decision"
      >
        Decision
      </div>
      <div
        className="Box-root Flex-root Flex-flex Flex-itemGutter gutter"
      >
        <button
          aria-label="Reject"
          className="BaseButton-root RejectButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="Icon-root Icon-lg RejectButton-icon"
          >
            close
          </span>
        </button>
        <button
          aria-label="Approve"
          className="BaseButton-root ApproveButton-root ApproveButton-invert"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="Icon-root Icon-lg ApproveButton-icon"
          >
            done
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rejects single comment 1`] = `
<div
  className="Card-root ModerateCard-root"
  data-testid="moderate-comment-comment-0"
>
  <div
    className="Box-root Flex-root Flex-flex"
  >
    <div
      className="ModerateCard-mainContainer"
    >
      <div
        className="ModerateCard-topBar"
      >
        <div
          className="Box-root Flex-root Flex-flex Flex-alignCenter"
        >
          <button
            className="BaseButton-root ModerateCard-username"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            <span
              className="Box-root Typography-root Typography-heading4 Typography-colorTextPrimary Username-root"
            >
              Isabelle
            </span>
          </button>
          <time
            className="RelativeTime-root Timestamp-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <button
            className="BaseButton-root FeatureButton-root"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            <span>
              Feature
            </span>
          </button>
        </div>
      </div>
      <div
        className="Box-root Typography-root Typography-bodyCopy Typography-colorTextPrimary ModerateCard-content CommentContent-root"
        dangerouslySetInnerHTML={
          Object {
            "__html": "This is the last random sentence I will be writing and I am going to stop mid-sent",
          }
        }
      />
      <div
        className="ModerateCard-footer"
      >
        <div
          className="Box-root HorizontalGutter-root HorizontalGutter-full"
        >
          <div>
            <a
              className="TextLink-root ModerateCard-link"
              href="http://localhost/comment/0"
              target="_blank"
            >
              View Context
              <span
                aria-hidden="true"
                className="Icon-root Icon-xs TextLink-icon"
              >
                open_in_new
              </span>
            </a>
          </div>
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="Box-root Flex-root Flex-flex Flex-itemGutter gutter"
            >
              <span
                className="Marker-root Marker-colorError Marker-variantRegular"
              >
                <span>
                  Spam
                </span>
                 
                <span
                  className="Count-root"
                >
                  2
                </span>
              </span>
              <button
                aria-controls="uuid-0"
                aria-expanded={false}
                className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Markers-detailsButtonColorRegular Button-variantRegular Markers-detailsButton"
                data-color="regular"
                data-variant="regular"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                type="button"
              >
                <span
                  className="Markers-detailsText"
                >
                  Details
                </span>
                <span
                  aria-hidden="true"
                  className="Icon-root Icon-sm"
                >
                  arrow_drop_down
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ModerateCard-separator ModerateCard-ruledSeparator"
    />
    <div
      className="Box-root Flex-root ModerateCard-aside ModerateCard-asideWithoutReplyTo Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionColumn gutter"
    >
      <div
        className="ModerateCard-decision"
      >
        Decision
      </div>
      <div
        className="Box-root Flex-root Flex-flex Flex-itemGutter gutter"
      >
        <button
          aria-label="Reject"
          className="BaseButton-root RejectButton-root RejectButton-invert"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="Icon-root Icon-lg RejectButton-icon"
          >
            close
          </span>
        </button>
        <button
          aria-label="Approve"
          className="BaseButton-root ApproveButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="Icon-root Icon-lg ApproveButton-icon"
          >
            done
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders single comment view 1`] = `
<div
  data-testid="single-moderate-container"
>
  <div
    className="SubBar-root SingleModerate-subBar SubBar-gutterBegin SubBar-gutterEnd"
  >
    <div
      className="Box-root Flex-root SubBar-container Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <a
        className="SingleModerate-subBarBegin"
        href="/admin/moderate/"
        onClick={[Function]}
      >
        Go to moderation queues
      </a>
      <div
        className="SingleModerate-subBarTitle"
      >
        Single Comment View
      </div>
    </div>
  </div>
  <div
    className="SingleModerate-background"
  />
  <div
    className="MainLayout-root"
  >
    <main
      className="SingleModerate-main"
    >
      <div
        className="Box-root HorizontalGutter-root Queue-root HorizontalGutter-double"
      >
        <div
          className="Card-root ModerateCard-root"
          data-testid="moderate-comment-comment-0"
        >
          <div
            className="Box-root Flex-root Flex-flex"
          >
            <div
              className="ModerateCard-mainContainer"
            >
              <div
                className="ModerateCard-topBar"
              >
                <div
                  className="Box-root Flex-root Flex-flex Flex-alignCenter"
                >
                  <button
                    className="BaseButton-root ModerateCard-username"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    type="button"
                  >
                    <span
                      className="Box-root Typography-root Typography-heading4 Typography-colorTextPrimary Username-root"
                    >
                      Isabelle
                    </span>
                  </button>
                  <time
                    className="RelativeTime-root Timestamp-root"
                    dateTime="2018-07-06T18:24:00.000Z"
                    title="2018-07-06T18:24:00.000Z"
                  >
                    2018-07-06T18:24:00.000Z
                  </time>
                  <button
                    className="BaseButton-root FeatureButton-root"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    type="button"
                  >
                    <span>
                      Feature
                    </span>
                  </button>
                </div>
              </div>
              <div
                className="Box-root Typography-root Typography-bodyCopy Typography-colorTextPrimary ModerateCard-content CommentContent-root"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "This is the last random sentence I will be writing and I am going to stop mid-sent",
                  }
                }
              />
              <div
                className="ModerateCard-footer"
              >
                <div
                  className="Box-root HorizontalGutter-root HorizontalGutter-full"
                >
                  <div>
                    <a
                      className="TextLink-root ModerateCard-link"
                      href="http://localhost/comment/0"
                      target="_blank"
                    >
                      View Context
                      <span
                        aria-hidden="true"
                        className="Icon-root Icon-xs TextLink-icon"
                      >
                        open_in_new
                      </span>
                    </a>
                  </div>
                  <div
                    className="Box-root HorizontalGutter-root HorizontalGutter-full"
                  >
                    <div
                      className="Box-root Flex-root Flex-flex Flex-itemGutter gutter"
                    >
                      <span
                        className="Marker-root Marker-colorError Marker-variantRegular"
                      >
                        <span>
                          Spam
                        </span>
                         
                        <span
                          className="Count-root"
                        >
                          2
                        </span>
                      </span>
                      <button
                        aria-controls="uuid-0"
                        aria-expanded={false}
                        className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Markers-detailsButtonColorRegular Button-variantRegular Markers-detailsButton"
                        data-color="regular"
                        data-variant="regular"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        type="button"
                      >
                        <span
                          className="Markers-detailsText"
                        >
                          Details
                        </span>
                        <span
                          aria-hidden="true"
                          className="Icon-root Icon-sm"
                        >
                          arrow_drop_down
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ModerateCard-separator ModerateCard-ruledSeparator"
            />
            <div
              className="Box-root Flex-root ModerateCard-aside ModerateCard-asideWithoutReplyTo Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionColumn gutter"
            >
              <div
                className="ModerateCard-decision"
              >
                Decision
              </div>
              <div
                className="Box-root Flex-root Flex-flex Flex-itemGutter gutter"
              >
                <button
                  aria-label="Reject"
                  className="BaseButton-root RejectButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="Icon-root Icon-lg RejectButton-icon"
                  >
                    close
                  </span>
                </button>
                <button
                  aria-label="Approve"
                  className="BaseButton-root ApproveButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="Icon-root Icon-lg ApproveButton-icon"
                  >
                    done
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
